DP-51: ajout de l'entité Tag

bilalkolli opened the pull request:
[DP-51](https://emjraffalli-1753106990807.atlassian.net/browse/DP-51)

- Création de l’entité Tag
- Création de la mutation createTag()
- Création de la query getAllTags

bilalkolli committed:
DP-51: ajout de l'entité Tag
>>
>> Création de l'entité Tag avec les champs : id, name, description, createdAt, updatedAt
>> Ajout de la mutation createTag() dans TagResolver
>> Ajout de la query getAllTags() pour récupérer tous les tags

Emilia-Raffalli commented on backend/src/inputs/post/TagInput.ts:
```diff
@@ -0,0 +1,10 @@
+
+@InputType()
+export class TagInput {
+  @Field()
```
Tu peux ajouter des validateurs avec class-validator pour une taille minimale re caractère requise par exemple sur tes champs ( voir SignupInput ou LoginInput)

Emilia-Raffalli commented on backend/src/resolvers/TagResolver.ts:
```diff
@@ -0,0 +1,28 @@
+
+@Resolver(Tag)
+export class TagResolver {
+@Mutation(() => Tag)
```
Attention à tes indentations

bilalkolli committed:
fix: ajout de validateurs dans TagInput et correction de l'indentation dans TagResolver

Emilia-Raffalli commented on backend/src/inputs/post/TagInput.ts:
```diff
@@ -0,0 +1,15 @@
+@InputType()
+export class TagInput {
+  @Field()
+  @MinLength(2, { message: "Le nom doit contenir au moins 2 caractères" })
```
Je pense que tu pourrais même plutôt faire un truc du genre 
@Lenght(2, 50) entre 2 et 50 caractères, (10 c'est un peu juste, nan ?)

plutôt que de mettre deux lignes @MinLength() et @MaxLenght()

`  @Length(2, 50, {message: "Le nom doit contenir entre 2 et 50 caractères"})
`  
 D'ailleurs je suis en train de me dire qu'un tag n'a pas besoin de description et que l'on peut supprimer la propriété, c'est juste un nom. Tu en penses quoi ?

bilalkolli committed:
refactor(tag): simplification de l'entité Tag

- Suppression du champ description (inutile pour un tag simple)
- Remplacement de @MinLength et @MaxLength par @Length dans TagInput

bilalkolli committed:
refactor(tag): ajustement des espacement dans les listes
